### ISSUE TITLE
[BUG]: Rogue Galaxy not rendering text characters after grayscale filter is used

### ISSUE THREAD

**Issue by piorrro33** [PCSX2/pcsx2#5258]
### Describe the Bug

Rogue Galaxy is missing text characters after grayscale filter is used. This only applies to Director's Cut and PAL, as NTSC-J (the first release) and NTSC-U render text differently.

### Reproduction Steps

1. Trigger a flashback event (easiest one to reach is new game -> play all the way to when you rent a yago and leave Rosa -> skip three cutscenes)
2. Skip the flashback event or play it to the end
3. Most characters are now missing

You can check if characters are missing or not by opening the game menu by pressing triangle and checking the location at the top of the screen. If there's nothing, well text is gone. You can also open the items menu and all item descriptions will be gone.

### Expected Behavior

Characters being there

### PCSX2 Revision

v1.7.2187

### Operating System

Windows 10 (64bit)

### CPU

i7-4790

### GPU

GTX 1060

### GS Settings

OpenGL, all default.

### Emulation Settings

Safest, but doesn't matter here

### GS Window Screenshots

Normal:

![gs_20220103152931](https://user-images.githubusercontent.com/6281092/147942345-0d62ae01-bcc3-43fa-b05c-48ce41033d4b.png)

Broken:

![gs_20220103153023](https://user-images.githubusercontent.com/6281092/147942419-993f4ac5-2962-46cc-a944-30dd6b32f253.png)

### Logs & Dumps

[GSDump of broken text](https://github.com/PCSX2/pcsx2/files/7802430/gsdump.zip)
[GSDump of normal text](https://github.com/PCSX2/pcsx2/files/7802434/gsdump_normal.zip)

**Comment by piorrro33** [comment]
I'll also add that there are two workarounds: **using the SW renderer** when the grayscale filter is used or enabling the **Fast Texture Invalidation** HW hack.

**Comment by RedDevilus** [comment]
Did you use a savestate?

**Comment by piorrro33** [comment]
Yes I did, but if needed I can go back to the flashback event without using savestates.

**Comment by RedDevilus** [comment]
Rogue Galaxy has a tendency to remove the text completely with savestates, it could be a different issue though. Just wanting to make sure.

**Comment by piorrro33** [comment]
Well, I believe these issues are the same or, at the very least, are linked between them. Could be cool to have actual reproduction steps for the savestate issue to add it to this issue or open a new one if appropriate.

Also indeed text is completely gone in PAL (in item descriptions et al.) but in Director's Cut, it's only some characters and even some parts of characters. Fast Texture Invalidation "fixes" both of them.

**Comment by MrCK1** [comment]
Possibly there may be more to this then just GS issue. A blockdump and savestate would probably be more useful here.

 The broken dump is still broken as of v1.7.4933. The other one is fine

**Comment by piorrro33** [comment]
Hi, unfortunately I didn't get the change to play with PCSX2 in a while.
I'll get to it ASAP, but in the meantime, I posted repro steps in the opening post. If you need better repro steps, let me know.

Thanks!

**Comment by Mrlinkwii** [comment]
What's the status of this issue on latest nightly
